### PR TITLE
GH-15780 set weak learner parameter API

### PR DESCRIPTION
**Pull request by valenad1**
#15780

### CODE REVIEW

**Review by wendycwong — DISMISSED** [review]

File: h2o-py/h2o/estimators/adaboost.py [hunk 1]
```diff
@@ -203,6 +214,28 @@ def learn_rate(self, learn_rate):
         assert_is_type(learn_rate, None, numeric)
         self._parms["learn_rate"] = learn_rate
 
+    @property
+    def weak_learner_params(self):
```
Comment (maurever) [inline h2o-py/h2o/estimators/adaboost.py:218]: Could you add an example here, please?

File: h2o-r/h2o-package/R/adaboost.R [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #' @param weak_learner Choose a weak learner type. Defaults to AUTO, which means DRF. Must be one of: "AUTO", "DRF", "GLM", "GBM",
 #'        "DEEP_LEARNING". Defaults to AUTO.
 #' @param learn_rate Learning rate (from 0.0 to 1.0) Defaults to 0.5.
+#' @param weak_learner_params Customized parameters for the weak_learner algorithm.
```
Comment (maurever) [inline h2o-r/h2o-package/R/adaboost.R:31]: Can you add example input here? Can you test if the input is correct?

Comment (valenad1) [reply h2o-r/h2o-package/R/adaboost.R:31]: Yes, I will add it.

R test is here: h2o-r/tests/testdir_algos/adaboost/runit_adaboost_weak_learner_smoke.R

Comment (valenad1) [reply h2o-r/h2o-package/R/adaboost.R]: Done

File: h2o-py/h2o/estimators/adaboost.py (hunk 1 above)
Comment (valenad1) [reply h2o-py/h2o/estimators/adaboost.py:218]: Done

**Review by maurever — APPROVED** [review]
LGTM, thanks @valenad1 !

**Review by wendycwong — APPROVED** [review]

**Pull request merged**